test/checkbox.test.tsx

```diff
@@ -0,0 +1,93 @@
+  )
+  .child("test `filter` by disabled", (test) =>
+    test
+      .step("render", () => render(<FormControlLabel label="disabled" disabled control={<Checkbox />} />))
+      .assertion(Interactor({ disabled: true }).exists())
+  )
+  .child("test `filter` by visible", (test) =>
+    test.step("render", () => render(<Checkbox />)).assertion(Interactor({ visible: false }).exists())
+  )
+  .child("test `click` action", (test) =>
+    test
+      .step("render", () => render(<FormControlLabel label="checkbox" control={<Checkbox />} />))
+      .assertion(Interactor().is({ checked: false, focused: false }))
+      .child("click and assert", (test) =>
+        test
+          .step(Interactor().click())
+          // TODO `focused` doesn't set :(
+          .assertion(Interactor().is({ checked: true /*, focused: true */ }))
+      )
+  )
+  .child("test `focus` action", (test) =>
+    test
+      .step("render", () => render(<FormControlLabel label="checkbox" control={<Checkbox />} />))
+      .assertion(Interactor().is({ focused: false }))
+      .child("focus and assert", (test) =>
+        test.step(Interactor().focus()).assertion(Interactor().is({ focused: true }))
+      )
+  )
+  .child("test `blur` action", (test) =>
+    test
+      .step("render", () => render(<FormControlLabel label="checkbox" control={<Checkbox autoFocus />} />))
+      .assertion(Interactor().is({ focused: true }))
+      .child("blur and assert", (test) => test.step(Interactor().blur()).assertion(Interactor().is({ focused: false })))
+  )
+  .child("test `check` action", (test) =>
+    test
+      .step("render", () => render(<FormControlLabel label="checkbox" control={<Checkbox />} />))
+      .assertion(Interactor().is({ checked: false }))
+      .child("check and assert", (test) =>
+        test.step(Interactor().check()).assertion(Interactor().is({ checked: true }))
+      )
+  )
+  .child("test `uncheck` action", (test) =>
+    test
+      .step("render", () => render(<FormControlLabel label="checkbox" control={<Checkbox defaultChecked />} />))
+      .assertion(Interactor().is({ checked: true }))
+      .child("uncheck and assert", (test) =>
+        test.step(Interactor().uncheck()).assertion(Interactor().is({ checked: false }))
+      )
+  )
+  // NOTE: Here is nice example of duplicating tests by useless parallelism
+  .child("test `toggle` action", (test) =>
+    test
+      .step("render", () => render(<FormControlLabel label="checkbox" control={<Checkbox />} />))
+      .assertion(Interactor().is({ checked: false }))
+      .child("toggle twice", (test) =>
+        test
+          .step(Interactor().toggle())
+          .assertion(Interactor().is({ checked: true }))
+          .child("second toggle", (test) =>
+            test.step(Interactor().toggle()).assertion(Interactor().is({ checked: false }))
+          )
+      )
+  );
+// TODO Don't work because `click` doesn't affect to `document.activeElement`
```

Comment:
Do we feel like we have a solution here? Or will it require an upstream change to form fields

Reply:
I'm going to make it here through the dispatch event. And later, we will make more accurate emulation for click in an upstream

Reply:
sounds like a good plan 👍🏻